### PR TITLE
Mark unread and jump to unread

### PR DESCRIPTION
**Pull request by martinmitrevski**
### 🔗 Issue Link
_Jira or Github issue link, if applicable._

### 🎯 Goal

_Describe why we are making this change._

### 🛠 Implementation

_Provide a description of the implementation._

### 🧪 Testing

_Describe the steps how this change can be tested (or why it can't be tested)._

### 🎨 Changes

_Add relevant screenshots or videos showcasing the changes._

### ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: Sources/StreamChatSwiftUI/ChatChannel/ChatChannelViewModel.swift [hunk 1]
```diff
@@ -198,7 +208,9 @@ open class ChatChannelViewModel: ObservableObject, MessagesDataSource {
     @objc
     private func applicationWillEnterForeground() {
         guard let first = messages.first else { return }
-        maybeSendReadEvent(for: first)
+        if canMarkRead {
+            maybeSendReadEvent(for: first)
```
Comment (testableapple) [inline Sources/StreamChatSwiftUI/ChatChannel/ChatChannelViewModel.swift:212]: Also, what do you think about renaming `maybeSendReadEvent()` and `maybeRefreshMessageList()` to `sendReadEventIfNeeded()` and `refreshMessageListIfNeeded()` accordingly?

File: Sources/StreamChatSwiftUI/ChatChannel/ChatChannelViewModel.swift [hunk 2]
```diff
@@ -552,6 +600,19 @@ open class ChatChannelViewModel: ObservableObject, MessagesDataSource {
...
+    private func checkUnreadCount() {
+        guard !isMessageThread else { return }
+        if channelController.channel?.unreadCount.messages ?? 0 > 0 {
+            if channelController.firstUnreadMessageId != nil {
+                firstUnreadMessageId = channelController.firstUnreadMessageId
+                canMarkRead = false
+            } else if channelController.lastReadMessageId != nil {
+                lastReadMessageId = channelController.lastReadMessageId
+                canMarkRead = false
+            }
```
[the remark this answers is not in the document]
Comment (martinmitrevski) [reply Sources/StreamChatSwiftUI/ChatChannel/ChatChannelViewModel.swift:612]: no, we should change the values only in these cases.

File: Sources/StreamChatSwiftUI/ChatChannel/ChatChannelViewModel.swift (hunk 1 above)
Comment (martinmitrevski) [reply Sources/StreamChatSwiftUI/ChatChannel/ChatChannelViewModel.swift]: done

File: Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageListView.swift [hunk 3]
```diff
@@ -287,6 +288,18 @@ public struct MessageListView<Factory: ViewFactory>: View, KeyboardReadable {
                 pendingKeyboardUpdate = nil
             }
         })
+        .overlay(
+            (channel.unreadCount.messages > 0 && !unreadMessagesBannerShown) ? 
+            factory.makeJumpToUnreadButton(
+                channel: channel,
+                onJumpToMessage: {
+                    _ = onJumpToMessage?(firstUnreadMessageId ?? "unknown")
```
[the remark this answers is not in the document]
Comment (martinmitrevski) [reply Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageListView.swift]: done

**Review by testableapple — APPROVED** [review]
LGTM 👍

**Pull request merged**